cirq/study/resolver.py

```diff
@@ -82,9 +85,15 @@ def value_of(self, value: Union['cirq.TParamKey', float]) -> 'cirq.TParamVal':
 
         Args:
             value: The parameter to try to resolve.
+            recursive: Whether to recursively evaluate formulas. If this is
+                set to True, the evaluation tree will be memoized in
+                self._deep_eval_map.
 
         Returns:
             The value of the parameter as resolved by this resolver.
+
+        Raises:
+            RecursionError if the ParamResolver detects a loop in resolution.
```

Comment:
Maybe "...a loop in a recursive resolution." to make it clear this cannot happen when `recursive=False`.